sphinx/writers/manpage.py

```diff
@@ -306,15 +305,23 @@ def visit_image(self, node: Element) -> None:
         self.body.append(_('[image]') + '\n')
         raise nodes.SkipNode
 
+
+    # Signal our support for this feature - clients can probe for this
+    # attribute to determine if we will emit OSC 8 markup.
+    HAVE_GROFF_OSC_8 = object()
```

Comment:
When would this object be changed? and how could it be used? here it's only a field that does not seem to be handled. 

Instead, why not using a configuration value where the client could decide whether OSC 8 markup is emitted?

Reply:
Also what happens when a terminal does *not* support those codes? would the output be messed up?

Reply:
> Also what happens when a terminal does not support those codes? would the output be messed up?

These escape codes follow the [OSC](https://en.wikipedia.org/wiki/ANSI_escape_code#OSC) pattern. AFAICT many applications already use another OSC command to set the window title. I'd be very surprised if there was a real-world terminal that reacted poorly to OSC 8 even if it doesn't know about OSC 8 specifically.

`groff` initially disabled it by default but [enabled it later](http://git.savannah.gnu.org/cgit/groff.git/commit/?id=9dadd72a3c9cd40c20e06e80e918a68badf83b0f)
This means users/distributions have another knob if they really want to disable - they can do so in groff, no need to know about Sphinx.

I wonder if we can find any use case where this hurts.

Besides terminals, I tried on old `groff` which simply ignores the unknown `\X` sequence.

> When would this object be changed? and how could it be used? here it's only a field that does not seem to be handled.

So far I only needed to detect whether this feature exists, not actually toggle it.

> Instead, why not using a configuration value where the client could decide whether OSC 8 markup is emitted?

I'm fine with a configuration value - I'd suggest enabling it by default. Or no configuration at all, I'm not sure why anyone would want to configure that.
Can I query for existence of the configuration value in my `conf.py`?

Alternatively I could check the Sphinx version instead, I guess that's less surprising than a magic name. I'm happy with that.

Reply:
Oh you mean to say that the sentinel object is used as a "versioning". Yes, I think checking the version is definitely better rather than this magical name (for me it was like a flag !)